Source/AliveLibAO/Abe.cpp

```diff
@@ -7885,9 +7887,322 @@ void Abe::State_87_428FA0()
+    switch (field_110_state)
+    {
+        case StoneStates::eUnknown_0:
+        {
+            if (field_10_anim.field_4_flags.Get(AnimFlags::eBit12_ForwardLoopCompleted))
+            {
+                ResourceManager::GetLoadedResource_4554F0(ResourceManager::Resource_Animation, 316, 1, 0);
+                field_164_pCircularFade = Make_Circular_Fade_447640(
+                    field_A8_xpos,
+                    field_AC_ypos,
+                    field_BC_sprite_scale,
+                    1,
+                    0
+                );
+
+                if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX))
+                {
+                    field_164_pCircularFade->field_10_anim.field_4_flags.Set(AnimFlags::eBit5_FlipX);
+                }
+                else
+                {
+                    field_164_pCircularFade->field_10_anim.field_4_flags.Clear(AnimFlags::eBit5_FlipX);
+                }
+
+                field_110_state = StoneStates::eUnknown_1;
+                SFX_Play_43AD70(SoundEffect::IngameTransition_107, 90, 0);
+                field_F0_pTlv = gMap_507BA8.TLV_Get_At_446260(
+                    FP_GetExponent(field_A8_xpos),
+                    FP_GetExponent(field_AC_ypos),
+                    FP_GetExponent(field_A8_xpos),
+                    FP_GetExponent(field_AC_ypos),
+                    TlvTypes::DemoPlaybackStone_96
+                );
+                if (!field_F0_pTlv)
+                    field_F0_pTlv = gMap_507BA8.TLV_Get_At_446260(
+                        FP_GetExponent(field_A8_xpos),
+                        FP_GetExponent(field_AC_ypos),
+                        FP_GetExponent(field_A8_xpos),
+                        FP_GetExponent(field_AC_ypos),
+                        TlvTypes::BellSongStone_54
+                    );
+                if (!field_F0_pTlv)
+                {
+                    field_F0_pTlv = gMap_507BA8.TLV_Get_At_446260(
+                        FP_GetExponent(field_A8_xpos),
+                        FP_GetExponent(field_AC_ypos),
+                        FP_GetExponent(field_A8_xpos),
+                        FP_GetExponent(field_AC_ypos),
+                        TlvTypes::MovieStone_51
+                    );
+                    sAbeSound_507730 = SFX_Play_43AE60(SoundEffect::HandstoneTransition_13, 127, -300, 0);
+                    if (!field_F0_pTlv)
+                        field_F0_pTlv = gMap_507BA8.TLV_Get_At_446260(
+                            FP_GetExponent(field_A8_xpos),
+                            FP_GetExponent(field_AC_ypos),
+                            FP_GetExponent(field_A8_xpos),
+                            FP_GetExponent(field_AC_ypos),
+                            TlvTypes::HandStone_100
+                        );
+                }
+
+                auto ptlv = static_cast<Path_Stone*>(field_F0_pTlv);
+                if (ptlv)
+                {
+                    field_174_pathStone = ptlv->field_18_data;
```

Comment:
the OG reads memory out of bounds here, we should probably fix it by switching on field_F0_pTlv->type and copying the corresponding sub struct of the union?